### PR TITLE
Fixed Phalcon\Config->merge not merging config with numeric properties properly

### PR DESCRIPTION
**Pull request by CameronHall**
Hello!

* Type: bug fix
* Link to issue: #13351

**In raising this pull request, I confirm the following (please check boxes):**

- [X] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)?
- [X] I have checked that another pull request for this purpose does not exist.
- [X] I wrote some tests for this PR.

Small description of change: Numeric config properties can now be merged properly.

Thanks

_4.0.x PR will follow shortly_

### CODE REVIEW

File: CHANGELOG-3.4.md [hunk 1]
```diff
@@ -43,6 +44,8 @@
 - Changed `Phalcon\Crypt::setCipher` so that IV length will be reconfigured during setting the cipher algorithm
 - Changed `Phalcon\Crypt::setCipher` so that method will throw `Phalcon\Crypt\Exception` if a cipher is unavailable
 - Fixed regression ([#13308](https://github.com/phalcon/cphalcon/pull/13308)) for `Phalcon\Debug\Dump::output` to correctly work with detailed mode [#13315](https://github.com/phalcon/cphalcon/issues/13315)
+- Do not throw Exception when superglobal does not exist [#13252](https://github.com/phalcon/cphalcon/issues/13252), [#13254](https://github.com/phalcon/cphalcon/issues/13254), [#12918](https://github.com/phalcon/cphalcon/issues/12918),
+[#13351](https://github.com/phalcon/cphalcon/issues/13351)
```
Comment (sergeyklay) [inline CHANGELOG-3.4.md]: @CameronHall Could you please sort out with change log? There is unrelated entries

**Review by sergeyklay — CHANGES_REQUESTED** [review]

File: CHANGELOG-3.4.md (hunk 1 above)
Comment (CameronHall) [reply CHANGELOG-3.4.md]: All done :) It's probably worthwhile asking, is there any way to amend a commit (to keep the git history clean) without any ci? Should I be adding `[ci skip]` to the description if I'm force-pushing an amendment like this?

Does `[ci skip]` skip Appveyor as well?

**Comment by CameronHall** [comment]
@sergeyklay all done :)

**Review by sergeyklay — DISMISSED** [review]

**Comment by sergeyklay** [comment]
@CameronHall I'm sorry. Could you please please rebase?

**Comment by CameronHall** [comment]
All done @sergeyklay

**Review by sergeyklay — APPROVED** [review]

**Comment by sergeyklay** [comment]
Thank you

**Pull request merged**